includes/Core/Util/Feature_Flags.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Util\Feature_Flags
+ *
+ * @package   Google\Site_Kit\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+use ArrayAccess;
+
+/**
+ * Class handling plugin activation.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Feature_Flags {
+	/**
+	 * Main instance for static access.
+	 *
+	 * @since n.e.x.t
+	 * @var Feature_Flags
+	 */
+	protected static $instance;
+
+	/**
+	 * Feature flag definitions.
+	 *
+	 * @since n.e.x.t
+	 * @var array|ArrayAccess
+	 */
+	private $features;
+
+	/**
+	 * Feature flag mode.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	private $mode;
+
+	/**
+	 * Gets the main feature flags instance.
+	 *
+	 * @since n.e.x.t
+	 * @internal
+	 *
+	 * @return Feature_Flags
+	 */
+	public static function get_instance() {
+		return static::$instance;
+	}
+
+	/**
+	 * Sets the main feature flags instance.
+	 *
+	 * @since n.e.x.t
+	 * @internal
+	 *
+	 * @param Feature_Flags $instance Feature_Flags instance to set as the main instance.
+	 */
+	public static function set_instance( Feature_Flags $instance ) {
+		static::$instance = $instance;
+	}
+
+	/**
+	 * Checks if the given feature is enabled in the current mode on the main instance.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $feature Feature key path to check.
+	 * @return bool
+	 */
+	public static function enabled( $feature ) {
+		return static::get_instance()->is_feature_enabled( $feature );
```

Comment:
When called too early, this could cause a fatal. Maybe we should cater for that and return `false` and possibly a `_doing_it_wrong`.